server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -556,6 +557,10 @@ protected RemoteStoreStatsTrackerFactory getRemoteStoreStatsTrackerFactory() {
         return remoteStoreStatsTrackerFactory;
     }
 
+    public String getNodeId() {
+        return translogConfig.getNodeId();
+    }
+
```

Comment:
Can't we get node id from ShardRouting#currentNodeId()

Reply:
The shard can be relocating to this node as well . So current node can be `ShardRouting#relocatingNodeId` as well .